src/Aurora.Shared/Models/ValueOrNull.cs

```diff
@@ -34,6 +34,20 @@ public TResult Resolve<TResult>(Func<T, TResult> onValue, Func<string, TResult>
         return result;
     }
 
+    public Task ResolveAsync(Func<T, Task> onValue, Func<string, Task> onNull)
+    {
+        Task result;
+        if (HasValue)
```

Comment:
use result.Then (you can add Then method like js has)

Reply:
introduced